windows/portaudio/JackPortAudioDriver.cpp

```diff
@@ -196,8 +196,12 @@ int JackPortAudioDriver::Open(jack_nframes_t buffer_size,
     }
 
     // Generic JackAudioDriver Open
+    char capture_driver_name[JACK_CLIENT_NAME_SIZE];
+    char playback_driver_name[JACK_CLIENT_NAME_SIZE];
+    snprintf(capture_driver_name, sizeof(capture_driver_name), "%s", capture_driver_uid);
+    snprintf(playback_driver_name, sizeof(playback_driver_name), "%s", playback_driver_uid);
```

Comment:
Thank you for your comment.

After the survey, I understand that this problem is more complicated than I thought... Let me summarize it (related to #182):

- In the specification, `snprintf()` always generates null-terminated results.
- `snprintf()` behaves correctly in most platforms, including MSVC >= 2015 and MinGW64.  MinGW64 uses msvcrt.dll but has own implementation.
- MSVC < 2015 does not have `snprintf()` but `_snprintf()`, which does NOT always generate null-terminated results.

and Jack source code does:

- `#define snprintf _snprintf` when the compiler is MSVC.
- The platform independent code seems to assume that `snprintf()` generates null-terminated results.
- The Windows dependent code seems to be mixed.  Some assumes null-terminated, and the other assumes non-null-termimanted.

So my proposal is: 

- Just remove `#define snprintf _snprintf` or make a thin wrapper for MSVC < 2015 to conform the specification.

- Eventually fixes the code that assumes non-null-terminated (Assuming non-null-terminated does not cause serious bugs in most cases, but result length may be shorter than expected).